src/range-to-string.js

```diff
@@ -0,0 +1,68 @@
+/* global Node */
+
+/**
+ * Return the next node after `node` in a tree order traversal of the document.
+ */
+function nextNode(node, skipChildren) {
+  if (!skipChildren && node.firstChild) {
+    return node.firstChild
+  } else if (node.nextSibling) {
+    return node.nextSibling
+  }
+  while (node.parentNode) {
+    node = node.parentNode
+    if (node.nextSibling) {
+      return node.nextSibling
+    }
+  }
+  /* istanbul ignore next */
+  return null
+}
+
+function firstNode(range) {
+  if (range.startContainer.nodeType === Node.ELEMENT_NODE) {
+    const node = range.startContainer.childNodes[range.startOffset]
+    return node || range.startContainer
```

Comment:
> If the offset is past all the children, that would logically place the contents of the range after the start container, to me. 

The most obvious case for this is when the range starts at an empty element, ie. `range.startContainer` is an element with no children (perhaps a `<br>` or empty paragraph or span) and `range.startOffset` is zero. I haven't verified whether it is possible to create such a range manually in any browser, but nothing in the spec forbids it either.

Returning `range.startContainer` here guarantees that the node returned by `firstNode(range)` <= `range.endContainer` (in tree order), which I think is a useful invariant to have.